test/api/unit_roles_test.rb

```diff
@@ -46,4 +62,68 @@ def test_post_unit_roles_not_unique
     assert_equal to_post[:unit_id], last_response_body['unit_id']
     assert_equal to_post[:user_id], last_response_body['user_id']
   end
+
+  # PUT tests
+  # Replace a unit role
+  def test_put_unit_role
+    # Details to replace
+    data_to_put = {
+      unit_role: FactoryGirl.build(:unit_role),
+      auth_token: auth_token
+    }
+
+    # Update unit_role with id = 1
+    put_json '/api/unit_roles/1', data_to_put
```

Comment:
you should get the id from the unit role

Reply:
If you build this from a factory then it already has this data in the database... so the put isn't doing anything. You will want to get the unit role, then do a put with different data... and check if the data has changed.

What aspect are you testing in this change?